airflow/providers/slack/hooks/slack.py

```diff
@@ -27,29 +28,50 @@
 # noinspection PyAbstractClass
 class SlackHook(BaseHook):
     """
+    Creates a Slack connection, to be used for calls.
     Takes both Slack API token directly and connection that has Slack API token.
-
     If both supplied, Slack API token will be used.
+    Exposes also the rest of slack.WebClient args.
 
     :param token: Slack API token
+    :type token: str
     :param slack_conn_id: connection that has Slack API token in the password field
+    :type slack_conn_id: str
+    :param use_session: A boolean specifying if the client should take advantage of
+        connection pooling. Default is True.
+    :type base_url: bool
+    :param base_url: A string representing the Slack API base URL. Default is
+        `https://www.slack.com/api/`
+    :type base_url: str
+    :param timeout: The maximum number of seconds the client will wait
+        to connect and receive a response from Slack.
+        Default is 30 seconds.
+    :type timeout: int
     """
-    def __init__(self, token: Optional[str] = None, slack_conn_id: Optional[str] = None) -> None:
+
+    def __init__(
+        self,
+        token: Optional[str] = None,
+        slack_conn_id: Optional[str] = None,
+        **client_args: Any,
+    ) -> None:
         super().__init__()
-        self.token = self.__get_token(token, slack_conn_id)
+        token = self.__get_token(token, slack_conn_id)
+        self.client = WebClient(token, **client_args)
```

Comment:
Let's not create a client in `__init__`

Reply:
Done.
I would assume it's more efficient to create the connection on init to possibly reuse. Do we have a common pattern in airflow hooks to create the connections during the call?

Reply:
I had a brain-fade moment actually 🤦 . I thought the client creation was inside Operator's init method, so to avoid DB calls while DAG Parsing I suggested to move it outside `__init__`. It won't matter with Hook

Reply:
Happy to merge it now as it is.

Reply:
I moved this back to init, to give the possibility to operators to use the client directly.